// @tomo-inc/uikit-lite/tailwind.config.js
import { colors, fontsizes } from "./src/styles/theme";

/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ["./src/**/*.{js,ts,jsx,tsx}"],
  darkMode: "selector",
  theme: {
    extend: {
      colors: {
        ...colors,
        primary: "var(--uikit-colors-primary)",
        secondary: "var(--uikit-colors-secondary)",
        // only primary btn
        primaryBtnText: {
          DEFAULT: "#fff",
          dark: "var(--uikit-colors-primary-btn-text-dark)",
        },
        bg1: "var(--uikit-background-BG1)",
        mbg: {
          DEFAULT: "#F6F6F6",
          dark: "#121212",
        },
        tc1: {
          DEFAULT: "#121212",
          dark: "#fff",
        },
        tc2: {
          DEFAULT: "#a0a0a0",
          dark: "#a0a0a0",
        },
        tc4: {
          DEFAULT: "#12121266",
          dark: "#ffffff66",
        },
        borc: {
          DEFAULT: "#f3f3f3",
          dark: "#303030",
        },
        border: "var(--uikit-colors-border)",
        border1: "var(--uikit-colors-border-1)",
        line: "var(--uikit-colors-line)",
        white: {
          DEFAULT: "#fff",
          dark: "#1b1b1b",
        },
        abg: {
          DEFAULT: "#D9D9D9",
          dark: "#444444",
        },
        iconc: "#999999",
        warning: "#FFBA0A",
        success: "#52c41a",
        danger: "#ED4A47",
      },
      boxShadow: {
        header: "0px 4px 4px 0px #D5D5D540",
      },
      fontFamily: {
        poppins: "Poppins",
        switzer: "Switzer",
        switzerLight: "SwitzerLight",
        switzerMedium: "SwitzerMedium",
        switzerBold: "SwitzerBold",
      },
      fontSize: {
        xs: "0.7rem",
      },
      borderRadius: {
        lg: "var(--uikit-rounded-lg)",
        DEFAULT: "var(--uikit-rounded)",
        md: "var(--uikit-rounded-md)",
        xl: "var(--uikit-rounded-xl)",
        "2xl": "var(--uikit-rounded-2xl)",
        "3xl": "var(--uikit-rounded-3xl)",
        full: "var(--uikit-rounded-full)",
      },
    },
    fontSize: {
      ...fontsizes,
    },
  },
  corePlugins: {
    preflight: false,
  },
  prefix: "uikit-",
  plugins: [],
};
