// @tomo-inc/uikit-lite/vite.config.ts
import react from "@vitejs/plugin-react";
import path from "path";
import { defineConfig } from "vite";
import dts from "vite-plugin-dts";
import { libInjectCss } from "vite-plugin-lib-inject-css";
import svgr from "vite-plugin-svgr";

// https://vite.dev/config/
export default defineConfig({
  plugins: [
    react(),
    svgr(),
    dts({
      include: ["src/**/*.ts", "src/**/*.tsx"],
      outDir: "dist",
      tsconfigPath: "./tsconfig.app.json",
    }),
    libInjectCss(),
  ],
  build: {
    lib: {
      entry: path.resolve(__dirname, "src/index.ts"),
      formats: ["umd", "cjs", "es"],
      name: "TmUikit",
      fileName: (format) =>
        format !== "es" ? `uikit.${format}.js` : "uikit.js",
    },
    rollupOptions: {
      external: [
        "react",
        "react/jsx-runtime",
        "react-router-dom",
        "react-dom",
        "classnames",
        "react-avatar",
        "qs",
        "axios",
        "react-toastify",
        "viem",
        "@wagmi/core",
        "wagmi",
        "crypto",
      ],
    },
    minify: false,
  },
  resolve: {
    alias: {
      "@": path.resolve(__dirname, "src"),
    },
  },
});
