// @tomo-inc/uikit-lite/src/styles/theme.ts
export const fontsizes = {
  n2: [
    "42px",
    {
      lineHeight: "1.2",
    },
  ],
  n3: [
    "36px",
    {
      lineHeight: "1.2",
    },
  ],
  h1: [
    "32px",
    {
      lineHeight: "1.2",
    },
  ],
  h2: [
    "28px",
    {
      lineHeight: "1.2",
    },
  ],
  h3: [
    "24px",
    {
      lineHeight: "1.4",
    },
  ],
  xl: [
    "20px",
    {
      lineHeight: "1.4",
    },
  ],
  base: [
    "16px",
    {
      lineHeight: "1.4",
    },
  ],
  sm: [
    "14px",
    {
      lineHeight: "1.4",
    },
  ],
  tiny: [
    "13px",
    {
      lineHeight: "1.2",
    },
  ],
  xs: [
    "12px",
    {
      lineHeight: "1.4",
    },
  ],
  df: [
    "10px",
    {
      lineHeight: "1.4",
    },
  ],
};

export const colors = {
  bg: "#0A090C",
  primary: "#f21f7f",
  card: "#1A1A1A",
  btn: "#000000",
  title: "#121212",
  desc: "#12121299",
  tc1: "#27272E",
  neutral4: "#E3E3E3",
  text3: "rgba(18, 18, 18, 0.60)",
  blue1: "#4285F4",
  blue2: "#1535CC",
  cardBg: "#EDF2F7",
  hint: "#7A828A",
  "red-1000": "#FF0042",
  "gray-1000": "#121212",
  "gray-600": "#12121299", //0.6
  "gray-400": "#12121266", //0.4
  "gray-200": "#12121233",
  "gray-100": "#F2F2F2",
  "gray-50": "#1212120D", //0.05
  "gray-25": "#12121206", //0.03
  "green-1000": "#2BBA81",
  "neutral-400": "#848795",
  "grey-18": "rgba(18, 18, 18, 0.40)",

  t1: "var(--uikit-text-t1)",
  t2: "var(--uikit-text-t2)",
  t3: "var(--uikit-text-t3)",
  t4: "var(--uikit-text-t4)",

  l1: "var(--uikit-background-l1)",
  bg1: "var(--uikit-background-BG1)",
  bg2: "var(--uikit-background-BG2)",
  bg3: "var(--uikit-background-BG3)",
  bg4: "var(--uikit-background-BG4)",
  bg5: "var(--uikit-background-BG5)",

  green: "#04C159",
  red: "#F21F7F",
  red2: "#EB4B6D",
  orange: "#FF9142",
  yellow: "#FFB904",
  blue: "#2B6BFF",
};

export const boxShadow = {
  toast: "0px 10px 32px 0px rgba(31, 31, 31, 0.15)",
};

export const spacings = {
  "13": "3.25rem",
};
